topsf/tasks/inference_v2/gen_toys.py

```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+
+import luigi
+import law
+import os
+
+from topsf.tasks.inference_v2.run_combine import RunCombine
+
+
+class GenToysV2(
+    RunCombine,
+):
+    n_toys = luigi.IntParameter(
+        default=-1,
+        significant=False,
+        description="Number of toys to generate",
+    )
+
+    set_parameters = law.CSVParameter(
+        significant=False,
+        description="Set parameters for the toys",
+    )
+
+    freeze_gen_parameters = law.CSVParameter(
+        significant=False,
+        description="Freeze parameters for the generation",
+    )
+
+    save_toys = luigi.BoolParameter(
+        default=True,
+        significant=False,
+        description="Save the generated toys",
+    )
+
+    gen_name = luigi.Parameter(
+        significant=True,
+        description="Name of the generated toys",
+    )
+
+    @property
+    def gen_toys_name(self):
+        name = f"gen_{self.gen_name}__{self.mode_inst}"
```

Comment:
It may be more informative to use `toys` as the base name here instead of `gen`.
```suggestion
        name = f"toys_{self.gen_name}__{self.mode_inst}"
```